ophidian/floorplan/Floorplan.h

```diff
@@ -0,0 +1,219 @@
+
+    //! Floorplan Constructor
+    /*!
+       \brief Constructs a floorplan system with no properties
+     */
+    Floorplan();
+
+    //! Floorplan Destructor
+    /*!
+       \brief Destroys the floorplan system, including its properties.
+     */
+    ~Floorplan();
+
+    //! Chip origin location setter
+    /*!
+       \brief Set the chip origin location.
+       \param loc Chip origin location.
+     */
+    void chipOrigin(util::Location loc);
+
+    //! Chip origin location getter
+    /*!
+       \brief Get the chip origin location.
+       \return Chip origin location.
+     */
+    util::Location chipOrigin()
+    {
+        return chipOrigin_;
+    }
+
+    //! Chip upper right corner location setter
+    /*!
+       \brief Set the chip upper right corner location.
+       \param loc Chip upper right corner location.
+     */
+    void chipUpperRightCorner(util::Location loc);
+
+    //! Chip upper right corner location getter
+    /*!
+       \brief Get the chip upper right corner location.
+       \param Chip upper right corner location.
+     */
+    util::Location chipUpperRightCorner()
+    {
+        return chipUpperRightCorner_;
+    }
+
+
+    // sites
+    //! Add site in the floorplan
+    /*!
+       \brief Adds a a new site in the floorplan. A site has a name and a dimension associated to it.
+       \param name Name of the site, used to identify it.
+       \param dimension Location describing the site dimension.
+       \return The created site.
+     */
+    Site add(Site, std::string name, util::Location loc);
+
+    //! Erase site in the floorplan
+    /*!
+       \brief Erases an existing site.
+       \param site Site to be erased.
+     */
+    void erase(Site site);
+
+
+    //! Name getter
+    /*!
+       \brief Get the name of a given site
+       \param site Site entity to get the name.
+       \return Name of the site.
+     */
+    std::string name(Site site) const
+    {
+        return names_[site];
+    }
+
+    //! Site Upper right corner getter
+    /*!
+       \brief get the upper right corner of a given site
+       \param site Site entity to get the upper right corner.
+       \return upper right corner of the site.
+     */
+    util::Location siteUpperRightCorner(Site site) const
+    {
+        return dimensions_[site];
+    }
+
+    //! Sites iterator
+    /*!
+       \return Range iterator for the Sites.
+     */
+    ophidian::util::Range<SitesIterator> sites_range() const
+    {
+        return ophidian::util::Range<SitesIterator>(sites_.begin(), sites_.end());
+    }
+
+
+    // rows
+    //! Add row in the floorplan
+    /*!
+       \brief Adds a a new row in the floorplan. A row has a origin location, number of sites and a siteTyoe associated to it.
+       \param loc Origin location of the row.
+       \param num Number of sites of the row.
+       \param site Site type of the row.
+       \return The created row.
+     */
+    Row add(Row, util::Location loc, size_t num, Site site);
```

Comment:
To keep the consistency, use const reference for `site` argument in this and similar cases.